pkg/trace/api/pipeline_stats.go

```diff
@@ -0,0 +1,102 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/logutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/fargate"
+)
+
+const (
+	pipelineStatsURLSuffix = "/api/v0.1/pipeline_stats"
+)
+
+// pipelineStatsEndpoint returns the pipeline intake url and the corresponding API key.
+func pipelineStatsEndpoint(cfg *config.AgentConfig) (url *url.URL, apiKey string, err error) {
+	if e := cfg.Endpoints; len(e) == 0 || e[0].Host == "" || e[0].APIKey == "" {
+		return nil, "", errors.New("config was not properly validated")
+	}
```

Comment:
Isn't this already validated elsewhere?

Reply:
It is validated here: https://github.com/DataDog/datadog-agent/blob/d277dfee88e4d522691e389c90b2910aa296164a/pkg/trace/writer/sender.go#L33

Do you think I should extract that into its own function?

Reply:
No, I think we're good here.